### PR TITLE
Update the roll dev script to actually work

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: dev/tools/lib/roll_dev.dart [hunk 1]
```diff
@@ -114,12 +143,12 @@ void main(List<String> args) {
   // Publish the archive before pushing the tag so that if something fails in
   // the publish step, we can clean up.
   try {
-    new ArchivePublisher(hash, version, Channel.dev)..publishArchive();
+    publisher.publishArchive();
   } on ArchivePublisherException catch (e) {
     print('Archive publishing failed.\n$e');
-    runGit('tag -d $version', 'remove the tag that was not published');
+    runGit('tag -d v$version', 'remove the tag that was not published');
```
Comment (gspencergoog) [inline dev/tools/lib/roll_dev.dart]: Whoa, good catch.

File: dev/tools/lib/roll_dev.dart [hunk 2]
```diff
@@ -16,13 +16,12 @@ const String kIncrement = 'increment';
...
 const String kY = 'y';
 const String kZ = 'z';
+const String kCommit = 'commit';
 const String kHelp = 'help';
 
-void main(List<String> args) {
-  // Set the cwd to the repo root, since we know where this script is located.
-  final Directory scriptLocation = new Directory(Platform.script.toFilePath());
-  Directory.current = scriptLocation.parent.parent.parent;
+const String kUpstreamRemote = 'git@github.com:flutter/flutter.git';
```
Comment (gspencergoog) [inline dev/tools/lib/roll_dev.dart]: What if they're using https access instead of .git access?

File: dev/tools/lib/roll_dev.dart (hunk 1 above)
Comment (Hixie) [reply dev/tools/lib/roll_dev.dart]: I'd actually forgotten about all of them, then when I submitted my PR to fix it it merge-conflicted with yours, and I missed this one when merging.

File: dev/tools/lib/roll_dev.dart (hunk 2 above)
Comment (Hixie) [reply dev/tools/lib/roll_dev.dart]: Our contributing docs say to use .git, I'm comfortable saying that if you want to publish you have to follow that script to the letter. :-)

Comment (gspencergoog) [reply dev/tools/lib/roll_dev.dart]: Works for me. :-)

**Review by gspencergoog — APPROVED** [review]

**Comment by gspencergoog** [comment]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**